azurerm/internal/services/synapse/tests/synapse_role_assignment_resource_test.go

```diff
@@ -0,0 +1,188 @@
+package tests
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/synapse/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func TestAccAzureRMSynapseRoleAssignment_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_synapse_role_assignment", "test")
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMSynapseRoleAssignmentDestroy,
+		Steps: []resource.TestStep{
+			// apply firewall rule first
+			{
+				Config: testAccAzureRMSynapseRoleAssignment_template(data),
+			},
+			// create synapse role assignment
+			{
+				Config: testAccAzureRMSynapseRoleAssignment_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMSynapseRoleAssignmentExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMSynapseRoleAssignment_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_synapse_role_assignment", "test")
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMSynapseRoleAssignmentDestroy,
+		Steps: []resource.TestStep{
+			// apply firewall rule first
+			{
+				Config: testAccAzureRMSynapseRoleAssignment_template(data),
+			},
+			// create synapse role assignment
+			{
+				Config: testAccAzureRMSynapseRoleAssignment_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMSynapseRoleAssignmentExists(data.ResourceName),
+				),
+			},
```

Comment:
how come this is a two step process? this means users won't be able to chain these togeather?

Reply:
user must have network access permission first and then they could assign synapse role. users could use depends_on to  chain these togeather

Reply:
Could we use depends on in the tf configs instead of 2 tests steps?